docs/src/main/paradox/jdbc.md

```diff
@@ -1,8 +1,225 @@
+
+## Dependencies
+
+To use the JDBC module of Akka Projections add the following dependency in your project:
+
+@@dependency [sbt,Maven,Gradle] {
+  group=com.lightbend.akka
+  artifact=akka-projection-jdbc_$scala.binary.version$
+  version=$project.version$
+}
+
+Akka Projections require Akka $akka.version$ or later, see @ref:[Akka version](overview.md#akka-version).
+
+@@project-info{ projectId="akka-projection-jdbc" }
+
+
+### Transitive dependencies
+
+The table below shows `akka-projection-jdbc`'s direct dependencies, and the second tab shows all libraries it depends on transitively.
+
+@@dependencies{ projectId="akka-projection-jdbc" }
+
+## Defining a JdbcSession
+
+> TODO - explain how and why we need one. Also the connection / thread pool relation. 
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #jdbc-session-imports #jdbc-session }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #todo }
 
 ## exactly-once
 
+The offset is stored in the same transaction used for the user defined `handler`, which means exactly-once
+processing semantics if the projection is restarted from previously stored offset.
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #projection-imports #exactlyOnce }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #exactlyOnce }
 
 ## at-least-once
+
+The offset is stored after the envelope has been processed and giving at-least-once processing semantics.
+This means that if the projection is restarted from a previously stored offset some elements may be processed more
+than once.
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #projection-imports #atLeastOnce }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #todo }
+
+The offset is stored after a time window, or limited by a number of envelopes, whatever happens first.
+This window can be defined with `withSaveOffset` of the returned `AtLeastOnceProjection`.
+The default settings for the window is defined in configuration section `akka.projection.at-least-once`.
+There is a performance benefit of not storing the offset too often, but the drawback is that there can be more
+duplicates when the projection that will be processed again when the projection is restarted.
+
+There is also an async variant `JdbcProjection.atLeastOnceAsync` to which you can pass an async `Handler` that returns a @scala[`Future`]@java[`ComplettionStage`]. Such an async handler doesn't store the projected result in the database, but it integrates with something else (eg: an Actor or some external system).   
```

Comment:
maybe better to put this in separate section as I did for Slick in https://github.com/akka/akka-projection/pull/274/files#diff-730eca1dd56a6ebcfec1a008e4349225